File: sample-app/src/main/java/org/granchi/whatnow/TasksModel.java
package org.granchi.whatnow;

import org.granchi.hollywood.Action;
import org.granchi.hollywood.Model;

import java.util.Collections;
import java.util.List;

import javax.inject.Inject;

/**
 * Model for existing tasks.
 *
 * @author serandel
 */
public class TasksModel extends Model {
    public final List<Task> tasks;

    @Inject
    public TasksModel() {
        this.tasks = Collections.emptyList();
    }

    @Override
    public Model actUpon(Action action) {
        return this;
    }
}

File: sample-app/src/main/kotlin/org/granchi/whatnow/TasksModel.kt
package org.granchi.whatnow

import org.granchi.hollywood.Action
import org.granchi.hollywood.Model
import javax.inject.Inject

/**
 * Model for existing tasks.
 */
class TasksModel @Inject constructor() : Model() {
    val tasks = emptyList<Task>()

    override fun actUpon(action: Action): Model? {
        return this
    }
}
